saged/remove_scrnaseq.py

```diff
@@ -0,0 +1,81 @@
+"""
+This script uses the recount metadata and a sparsity threshold to remove
+sc-RNAseq data from the recount compendium
+"""
+
+import argparse
+
+import pandas as pd
+import tqdm
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('count_file', help='The file containing the count matrix generated by '
+                                           'download_recount3.R')
+    parser.add_argument('metadata_file', help='The file with info about samples')
+    parser.add_argument('out_file', help='The file to save the normalized results to')
+    args = parser.parse_args()
+
+    metadata = pd.read_csv(args.metadata_file, sep='\t')
+    # Drop rows without sample ids and duplicate rows
+    metadata = metadata[metadata['external_id'].notna()]
+    metadata = metadata.drop_duplicates(subset=['external_id', 'study'])
+    # Set the index to the sample id for faster access
+    metadata = metadata.set_index('external_id')
+
+    LINES_IN_FILE = 317259
+
+    with open(args.count_file, 'r') as count_file:
+        out_file = open(args.out_file, 'w')
```

Comment:
Do you want to close this file handle somewhere?

Reply:
Good call